libevm/VM.cpp

```diff
@@ -119,22 +152,20 @@ void VM::adjustStack(unsigned _removed, unsigned _added)
 
 void VM::updateSSGas()
 {
-    if (!m_ext->store(m_SP[0]) && m_SP[1])
-        m_runGas = toInt63(m_schedule->sstoreSetGas);
-    else if (m_ext->store(m_SP[0]) && !m_SP[1])
-    {
-        m_runGas = toInt63(m_schedule->sstoreResetGas);
-        m_ext->sub.refunds += m_schedule->sstoreRefundGas;
-    }
+    evm_uint256be key = toEvmC(m_SP[0]);
+    evm_uint256be rawValue;
+    m_context->fn_table->get_storage(&rawValue, m_context, &m_message->destination, &key);
```

Comment:
When is `updateSSGas` called? Before applying the actual change? How was this correct before? And how is it correct now (it doesn't handle the refund case)?

Reply:
It's called at the beginning of SSTORE.
In EVM-C refunds are handled by the Client (in this case by ExtVMFace.cpp). Mostly because they are applied at the end of transaction and VM has only the "call" scope. Maybe it would not be so difficult to count refunds in VM and return it with evm_result.